src/components/metadata-modal/metadata-modal.js

```diff
@@ -46,7 +46,7 @@ const MetadataModal = ({ metadata, onToggle, visible }) => {
         </div>
         {hasPreview && (
           <div className="pipeline-metadata-modal__preview-text">
-            Previewing first 40 rows only
+            Previewing first {metadata.preview.data.length} rows only
```

Comment:
I think this copy line might not make sense anymore. We use it originally because we were only going to show the first 40 rows. But now since users can define it themself, do we need it anymore?

Reply:
I think it's still useful to have the message but maybe rephrase it to: "Previewing first ... rows"

Reply:
Agree with Merel here. Let's still give them the number.